### ISSUE TITLE
Bnd releases to first writable repo if -releaserepo is not set

### ISSUE THREAD

**Issue by juergen-albert** [bndtools/bnd#4395]
While trying to fix another issue I stumbled upon some behavior that raised a few question marks in my haead.

The `Project.getReleaseRepos(String names)` contains the following logic:
```
                if (repoNames == null) { // -releaserepo unspecified
			for (RepositoryPlugin plugin : plugins) {
				if (plugin.canWrite()) {
					result.add(plugin);
					break;
				}
			}
			if (result.isEmpty()) {
				msgs.NoNameForReleaseRepository();
			}
			return result;
		}
```
Trying a release without a defined release repo, should result in an error if you ask me, instead of taking the first writable repository we can find. I was a bit confused, as my release build where I accidentally removed the releaserepos instruction tried to upload to Maven Central.

If you ask me, this should result in an error or at least a warning to explain the behavior, because this is too much implicit knowledge one needs here.

**Comment by pkriens** [comment]
I think this was done for compatibility reasons a long time ago. I think documenting it is ok, you need to set the -releaserepo to empty to not have a release

**Comment by juergen-albert** [comment]
I'd like to add a warning as well, so users can see it in the build log as well without searching for it in the documentation.

**Comment by bjhargrave** [comment]
Fixed by #4448
